pandas/plotting/_core.py

```diff
@@ -287,6 +287,9 @@ def _maybe_right_yaxis(self, ax, axes_num):
 
             if not self._has_plotted_object(orig_ax):  # no data on left y
                 orig_ax.get_yaxis().set_visible(False)
+
+            if self.logy or self.loglog:
```

Comment:
Just FYI @charlesdong1991 we'll want to update https://github.com/pandas-dev/pandas/pull/24968 to handle this new case.

This PR is being back ported to 0.24.2. After it's in master, we'll want to think about consolidating all this "axis log" stuff.